### PR TITLE
Pull request for swift-retry-only-metricd

### PR DESCRIPTION
**Pull request by jd**
indexer: remove connect() method
Only retry connection to external components in metricd
utils: do not return coordinator id
cli: limit scope of error catching
cli: use MetricProcessBase _configure method to get drivers

### CODE REVIEW

**Comment by jd** [comment]
Pull-request updated, HEAD is now 1336644befd5576be3da5a57afed92bd12c4c31b

**Comment by chungg** [comment]
is the title of this PR wrong? i don't understand how this relates to swift retry?

**Comment by jd** [comment]
Haha yeah, the name is wrong. It's related to Swift but to retry only metricd, not API.

File: gnocchi/cli.py [hunk 1]
```diff
@@ -190,36 +190,21 @@ class MetricProcessor(MetricProcessBase):
...
             worker_id, conf, conf.metricd.metric_processing_delay)
-        self.coord, __ = utils.get_coordinator_and_start(
+        self.coord = utils.get_coordinator_and_start(
             conf.storage.coordination_url)
         self._tasks = []
         self.group_state = None
 
     @utils.retry
     def _configure(self):
-        self.store = storage.get_driver(self.conf, self.coord)
```
Comment (chungg) [inline gnocchi/cli.py]: oh now, i know why. i did this to reuse the coordinator for storage. i think we still want that?

 c307faa535a00b5525b73bb29a434ba2692f8c67

Comment (jd) [inline gnocchi/cli.py]: Oh my bad, I completely missed the self.coord being passed. I'll fix it.

**Comment by jd** [comment]
Something is wrong with my PR since it misses patches… sigh.

**Comment by jd** [comment]
Pull-request updated, HEAD is now d30d187923329d5ca00eb17ccaa1c975122bbbfb

**Comment by chungg** [comment]
@jd  that makes more sense now. i was wondering why your PR title said stuff about swift but all i saw in patches was stuff related to not passing coordinator.

**Comment by jd** [comment]
@chungg yeah it does not ironically… oslo.db connect and reconnect automatically, Gnocchi can't control anything, for good or bad reasons.

**Comment by jd** [comment]
Pull-request updated, HEAD is now 5103dc82b357e814ccd7248435ed247a2e04ced6

**Comment by jd** [comment]
Just removed the requirements change. The tenacity version listed as a dependency should be good enough.

**Comment by jd** [comment]
Rebased against master

**Pull request merged**